recipes/json-glib/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+from conans import ConanFile, tools, Meson, VisualStudioBuildEnvironment
+
+import os
+import glob
+import shutil
+
+required_conan_version = ">=1.29"
+
+class JsonGlibConan(ConanFile):
+    name = "json-glib"
+    description = "JSON-GLib implements a full JSON parser and generator using GLib and GObject, and integrates JSON with GLib data types."
+    topics = ("json", "glib")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://wiki.gnome.org/Projects/JsonGlib"
+    license = "GNU LGPL-2.1"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_introspection": ['enabled', 'auto', 'disabled'],
+        "gtk_doc": ['enabled', 'auto', 'disabled'],
+        "man": [True, False],
+        "tests": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_introspection": 'auto',
+        "gtk_doc": 'auto',
+        "man": False,
+        "tests": False,
+    }
+    generators = "pkg_config"
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        self.requires("glib/2.68.0")
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        # `tests' option was added in 1.5
+        if tools.Version(self.version) < '1.5':
+            del self.options.tests
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("meson/0.56.2")
+        if self.settings.os == 'Windows':
+            self.build_requires('pkgconf/1.7.4')
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        tools.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
```

Comment:
"new" conan 1.33 features
```suggestion
        tools.get(**self.conan_data["sources"][self.version],
                  destination=self._source_subfolder, strip_root=True)
```